### PR TITLE
[SYCL][E2E][LIT] Normalize devices not to include "ext_oneapi_"

### PR DESCRIPTION
**Pull request by aelovikov-intel**
Also see:
https://github.com/intel/llvm/pull/12007#discussion_r1405803885 https://github.com/intel/llvm/issues/12048

### CODE REVIEW

**Review by steffenlarsen — APPROVED** [review]
LGTM! 👍

**Review by sommerlukas — APPROVED** [review]
Changes to the kernel fusion test LGTM.

File: sycl/test-e2e/Graph/RecordReplay/buffer_copy_host2target_2d.cpp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // RUN: %{build} -o %t.out
 // RUN: %{run} %t.out
 // Extra run to check for leaks in Level Zero using UR_L0_LEAKS_DEBUG
-// RUN: %if ext_oneapi_level_zero %{env UR_L0_LEAKS_DEBUG=1 %{run} %t.out 2>&1 | FileCheck %s %}
+// RUN: %if level_zero %{env UR_L0_LEAKS_DEBUG=1 %{run} %t.out 2>&1 | FileCheck %s %}
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Graph/RecordReplay/buffer_copy_host2target_2d.cpp:4]: 2Graph team: I'd suggest to modify your RUN lines to use `FileCheck %s --implicit-check-not=LEAK` and drop line 6 here and in other tests.

**Review by EwanC — APPROVED** [review]

File: sycl/test-e2e/Graph/RecordReplay/buffer_copy_host2target_2d.cpp (hunk 1 above)
Comment (EwanC) [reply sycl/test-e2e/Graph/RecordReplay/buffer_copy_host2target_2d.cpp:4]: Thanks for the tip, we can do this as a follow-up PR

**Comment by aelovikov-intel** [comment]
@intel/dpcpp-tools-reviewers , ping.

**Review by sarnex — APPROVED** [review]

**Pull request merged**